### PR TITLE
expression: add builtin sha1

### PR DESCRIPTION
**Pull request by haoxiang47**
## What have you changed? (mandatory)

add ```sha1``` to function in ```builtin_encryption.rs```

## What are the type of the changes? (mandatory)

 * improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)

pass ```make expression```

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)

no

## Does this PR affect tidb-ansible update? (mandatory)

no

### CODE REVIEW

**Comment by breeswish** [comment]
Thanks! Please also add tests about null case. Rest LGTM

**Comment by haoxiang47** [comment]
@breeswish I had added the null case in test_sha1, just like test_md5 above. i just implement like test_md5 do.

File: src/coprocessor/dag/expr/builtin_encryption.rs [hunk 1]
```diff
@@ -29,6 +29,18 @@ impl ScalarFunc {
...
+
+    pub fn sha1<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &[Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let input = try_opt!(self.children[0].eval_string(ctx, row));
+        let mut hasher = Sha1::new();
+        hasher.input(input.as_ref());
+        let sha1 = hasher.result_str().into_bytes();
```
Comment (overvenus) [inline src/coprocessor/dag/expr/builtin_encryption.rs]: How about `Sha1::result()`?

Comment (haoxiang47) [reply src/coprocessor/dag/expr/builtin_encryption.rs]: I follow the rs doc with this usage, but I meet ```test_check_args failed```. 
I will use ``` Sha1::result()``` to have a try

**Comment by hicqu** [comment]
LGTM. Please address comments.

**Comment by haoxiang47** [comment]
@overvenus I have updated the api from result_str -> result

**Comment by overvenus** [comment]
Hm test failed 🙁

**Comment by TennyZhuang** [comment]
@hicqu `result` return the real memory view of the sha1, but we need hex encoded result, so that `result_str` is the best option. If we use `result`, we still need to hex encode it manually

**Comment by TennyZhuang** [comment]
Equivalent code in Python
image

**Comment by haoxiang47** [comment]
@overvenus @hicqu just as @TennyZhuang say, we think ```result_str``` is the best option, I had revert the code back to this implement.

**Comment by hicqu** [comment]
LGTM.

**Pull request merged**